pkg/cloudprovider/providers/openstack/openstack_instances.go

```diff
@@ -222,12 +223,18 @@ func srvInstanceType(srv *servers.Server) (string, error) {
 	return "", fmt.Errorf("flavor name/id not found")
 }
 
+// If Instances.InstanceID or cloudprovider.GetInstanceProviderID is changed, the regexp should be changed too.
+var providerIDRegexp = regexp.MustCompile(`^` + ProviderName + `:///([^/]+)$`)
```

Comment:
I've moved this outside of the func to make sure we compile it only once and not everytime `instanceIDFromProviderID` is called, also has the advantage that should this ever be wrong, we panic on binary start rather than when `instanceIDFromProviderID` is called for the first time